docs/advanced_topics/images/title_generation_on_upload.md

```diff
@@ -0,0 +1,107 @@
+# Title generation on upload
+
+Override how the title is set when adding a single image, multiple images or uploading an image within a chooser modal.
+
+When a file is dropped into the multi-upload page or selected on the single file selection form a title will be automatically populated into the Title field. The default behaviour is to use the image's filename excluding the extension.
+
+Using a JavaScript [event listener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) which will listen to the `'wagtail:images-upload'` event you can customise the resolved value of this title.
+
+The simplest way to add JavaScript to the editor is via the [`insert_global_admin_js` hook](../../reference/hooks.html#insert-global-admin-js), however any JavaScript that adds the event listener will work.
+
+## DOM Event
+
+The event name to listen for is `'wagtail:images-upload'` and it will be dispatched on the image upload `form`. The event's `detail` will contain a `data` object containing the `title` that will be used (filename without extension). `detail` will also include a `maxTitleLength` integer (or `null`) which will be the max character length on the Image model's title field and `filename`, the original filename without the extension removed.
```

Comment:
To allow mutation of the title it must be in a nested object, so hence the `data` containing `title`.

These other two parts are read only, hence directly on the `detail` object.
The `maxTitleLength` is there in case customisation wants to do something smart with reducing the title length.
The `filename` is there to provide access to the raw (with file extension) name.